### PR TITLE
core/validatorapi: add sync committee duties to validatorapi

### PR DESCRIPTION
**Pull request by dB2510**
Adds support for sync committee duties to validatorapi.

category: feature
ticket: #1261

### CODE REVIEW

File: core/validatorapi/eth2types.go [hunk 1]
```diff
@@ -135,3 +135,41 @@ func (v v1Validator) MarshalJSON() ([]byte, error) {
 
 	return bytes.ToLower(b), nil // ValidatorState must be lower case.
 }
+
+// syncCommitteeDutiesRequest defines the request to the getSyncCommitteeDuties endpoint.
+// See https://ethereum.github.io/beacon-APIs/#/ValidatorRequiredApi/getSyncCommitteeDuties.
+type syncCommitteeDutiesRequest []eth2p0.ValidatorIndex
```
Comment (corverroos) [inline core/validatorapi/eth2types.go:141]: rather just rename `attesterDutiesRequest` to `valIndexesJSON` and re-use that type.

File: core/validatorapi/router.go [hunk 2]
```diff
@@ -384,6 +390,32 @@ func attesterDuties(p eth2client.AttesterDutiesProvider) handlerFunc {
...
+			return nil, err
+		}
+
+		data, err := p.SyncCommitteeDuties(ctx, eth2p0.Epoch(epoch), req)
+		if err != nil {
+			return nil, err
+		}
+
+		if len(data) == 0 {
+			data = []*eth2v1.SyncCommitteeDuty{}
```
Comment (corverroos) [inline core/validatorapi/router.go:412]: is this really required? why isn't nil fine?

File: core/validatorapi/validatorapi.go [hunk 3]
```diff
@@ -822,6 +822,24 @@ func (c Component) AttesterDuties(ctx context.Context, epoch eth2p0.Epoch, valid
...
 }
 
+func (c Component) SyncCommitteeDuties(ctx context.Context, epoch eth2p0.Epoch, validatorIndices []eth2p0.ValidatorIndex) ([]*eth2v1.SyncCommitteeDuty, error) {
+	duties, err := c.eth2Cl.SyncCommitteeDuties(ctx, epoch, validatorIndices)
+	if err != nil {
+		return nil, err
+	}
+
+	// Replace root public keys with public shares.
+	for i := 0; i < len(duties); i++ {
```
Comment (corverroos) [inline core/validatorapi/validatorapi.go:832]: no test for this?

File: core/validatorapi/eth2types.go (hunk 1 above)
Comment (dB2510) [reply core/validatorapi/eth2types.go]: done

File: core/validatorapi/validatorapi.go (hunk 3 above)
Comment (dB2510) [reply core/validatorapi/validatorapi.go:832]: added

**Review by corverroos — APPROVED** [review]

**Pull request merged**